scripts/pylib/twister/twisterlib/coverage.py

```diff
@@ -330,7 +330,10 @@ def _generate(self, outdir, coveragelog):
         ztestfile = os.path.join(outdir, "ztest.json")
 
         excludes = Gcovr._interleave_list("-e", self.ignores)
-        excludes += Gcovr._interleave_list("--exclude-branches-by-pattern", self.ignore_branch_patterns)
+        if len(self.ignore_branch_patterns) > 0:
```

Comment:
Just `if self.ignore_branch_patterns:` would suffice and cover the case where this is `None`, but that situation should, in theory, never happen. Can stay as-is.

Reply:
It would suffice, but it is less informative regarding the expected type and is less useful IMO than `if len(array):`, which itself is picked up by various tools as being unclear